app/components/header.hbs

```diff
@@ -21,15 +21,29 @@
       {{#if this.session.currentUser}}
         <Dropdown data-test-user-menu as |dd|>
           <dd.Trigger local-class="dropdown-button" data-test-toggle>
-            <UserAvatar @user={{this.session.currentUser}} @size="small" local-class="avatar" data-test-avatar />
+            <UserAvatar @user={{this.session.currentUser}} @size="small" @sudo={{this.session.isSudoEnabled}} local-class="avatar" data-test-avatar />
             {{ this.session.currentUser.name }}
           </dd.Trigger>
 
-          <dd.Menu local-class="current-user-links" as |menu|>
-            <menu.Item><LinkTo @route="dashboard">Dashboard</LinkTo></menu.Item>
-            <menu.Item><LinkTo @route="settings" data-test-settings>Account Settings</LinkTo></menu.Item>
-            <menu.Item><LinkTo @route="me.pending-invites">Owner Invites</LinkTo></menu.Item>
-            <menu.Item local-class="menu-item-with-separator">
+          <dd.Menu local-class='current-user-links' as |menu|>
+            <menu.Item><LinkTo @route='dashboard'>Dashboard</LinkTo></menu.Item>
+            <menu.Item><LinkTo @route='settings' data-test-settings>Account Settings</LinkTo></menu.Item>
+            <menu.Item><LinkTo @route='me.pending-invites'>Owner Invites</LinkTo></menu.Item>
+            {{#if this.session.isAdmin}}
+              <menu.Item local-class='sudo'>
+                {{#if this.session.isSudoEnabled}}
```

Comment:
Does this update correctly when the sudo period is over? Since `isSudoEnabled` uses `Date.now()` instead of a "tracked" thing we might need something to tell the template to rerender/reevaluate every X seconds.

I think the way I would architect it to avoid the issue is by using http://ember-concurrency.com: we would have a restartable `sudoEnabledTask` on the `session` service with just `await rawTimeout(6 hours)` in it, and then we can use the derived property `this.sudoEnabledTask.isRunning` to check if sudo mode is enabled or not. that property should automatically cause the template to update once the task has finished (i.e. after the 6 hour timeout). to manually disable the sudo mode we can use `this.sudoEnabledTask.cancelAll()`.

Reply:
> Does this update correctly when the sudo period is over?

No, and I meant to put that in the PR description and clearly forgot when a giant golden retriever started pointing out that he was late for his game of fetch last night.

I like the task approach, so I've implemented that.